### ISSUE TITLE
custom_nodes\gguf\pig.py:342: UserWarning: The given NumPy array is not writable

### ISSUE THREAD

**Issue by offoxx** [calcuis/gguf#31]
Hello

I have a error in Comfy UI, few days ago I update python dependencies and update Comfy UI, now I have low performance in video generation with WAN 2.1, i reach 17 minutes per 53 frames in video generation before the updates: 

ComfyUI_windows_portable\ComfyUI\custom_nodes\gguf\pig.py:342: UserWarning: The given NumPy array is not writable, and PyTorch does not support non-writable tensors. This means writing to this tensor will result in undefined behavior. You may want to copy the array to protect its data or make it writable before converting it to a tensor. This type of warning will be suppressed for the rest of this program. (Triggered internally at C:\actions-runner\_work\pytorch\pytorch\pytorch\torch\csrc\utils\tensor_numpy.cpp:209.)
  torch_tensor = torch.from_numpy(tensor.data)

I have installed triton + sage attention v2 (2.1.1)
RTX 4060TI 16GB
CUDA 12.6

list of embedded python dependencies

Package                    Version      Editable project location
-------------------------- ------------ ------------------------------------------------------------------------------------------------------
absl-py                    2.2.2
accelerate                 1.6.0
addict                     2.4.0
aiohappyeyeballs           2.4.0
aiohttp                    3.11.18
aiosignal                  1.3.1
albucore                   0.0.16
albumentations             1.4.15
annotated-types            0.7.0
antlr4-python3-runtime     4.9.3
anyio                      4.9.0
attrs                      25.3.0
av                         14.3.0
beautifulsoup4             4.13.4
cachetools                 5.5.2
certifi                    2024.8.30
cffi                       1.17.1
chardet                    5.2.0
charset-normalizer         3.3.2
click                      8.1.8
color-matcher              0.6.0
colorama                   0.4.6
coloredlogs                15.0.1
colorlog                   6.9.0
colour-science             0.4.6
comfyui_frontend_package   1.18.10
comfyui_workflow_templates 0.1.14
contourpy                  1.3.2
cryptography               44.0.3
cssselect2                 0.8.0
cupy-cuda12x               13.4.1
cycler                     0.12.1
Cython                     3.1.0
ddt                        1.7.2
deepdiff                   8.5.0
Deprecated                 1.2.18
diffusers                  0.33.1
dill                       0.4.0
docutils                   0.21.2
easydict                   1.13
einops                     0.8.1
eval_type_backport         0.2.2
fastrlock                  0.8.3
filelock                   3.15.4
flatbuffers                25.2.10
flet                       0.28.2
fonttools                  4.58.0
frozenlist                 1.4.1
fsspec                     2024.9.0
ftfy                       6.3.1
gdown                      5.2.0
gguf                       0.16.3
gitdb                      4.0.12
GitPython                  3.1.44
h11                        0.16.0
httpcore                   1.0.9
httpx                      0.28.1
huggingface-hub            0.31.1
humanfriendly              10.0
idna                       3.8
imageio                    2.37.0
imageio-ffmpeg             0.6.0
importlib_metadata         8.7.0
iopath                     0.1.10
jax                        0.6.0
jaxlib                     0.6.0
Jinja2                     3.1.4
joblib                     1.5.0
jsonschema                 4.23.0
jsonschema-specifications  2025.4.1
kiwisolver                 1.4.8
kornia                     0.8.1
kornia_rs                  0.1.9
lark                       1.2.2
lazy_loader                0.4
llvmlite                   0.44.0
lxml                       5.4.0
manifold3d                 3.0.1
mapbox_earcut              1.0.3
markdown-it-py             3.0.0
MarkupSafe                 2.1.5
matplotlib                 3.10.3
matrix-client              0.4.0
mdurl                      0.1.2
ml_dtypes                  0.5.1
mpmath                     1.3.0
mss                        10.0.0
multidict                  6.0.5
networkx                   3.3
ninja                      1.11.1.4
numba                      0.61.2
numpy                      1.26.4
nvidia-cuda-runtime-cu12   12.9.37
nvidia-ml-py               12.575.51
oauthlib                   3.2.2
omegaconf                  2.3.0
onnxruntime-gpu            1.22.0
opencv-contrib-python      4.11.0.86
opencv-python              4.11.0.86
opencv-python-headless     4.11.0.86
opt_einsum                 3.4.0
orderly-set                5.4.1
packaging                  25.0
pandas                     2.2.3
piexif                     1.1.3
pillow                     11.0.0
pip                        25.1.1
pixeloe                    0.1.4
platformdirs               4.3.8
polygraphy                 0.49.20
pooch                      1.8.2
portalocker                3.1.1
propcache                  0.3.1
protobuf                   6.30.2
psutil                     7.0.0
py-cpuinfo                 9.0.0
pycollada                  0.9
pycparser                  2.22
pydantic                   2.11.4
pydantic_core              2.33.2
pygit2                     1.18.0
PyGithub                   2.6.1
Pygments                   2.19.1
PyJWT                      2.10.1
PyMatting                  1.1.13
PyNaCl                     1.5.0
pynvml                     12.0.0
pyparsing                  3.2.3
pyreadline3                3.5.4
PySocks                    1.7.1
python-dateutil            2.9.0.post0
pytz                       2025.2
pywin32                    310
PyYAML                     6.0.2
q8_kernels                 0.0.4
referencing                0.36.2
regex                      2024.7.24
rembg                      2.0.65
repath                     0.9.0
reportlab                  4.4.0
requests                   2.32.3
rich                       14.0.0
rpds-py                    0.24.0
rtree                      1.4.0
safetensors                0.5.3
sageattention              2.1.1        D:\ComfyUI_windows_portable_nvidia LTXV - copia - copia\ComfyUI_windows_portable\ComfyUI\SageAttention
scikit-image               0.25.2
scikit-learn               1.6.1
scipy                      1.15.3
seaborn                    0.13.2
segment-anything           1.0
sentencepiece              0.2.0
setuptools                 80.8.0
shapely                    2.1.0
shellingham                1.5.4
simsimd                    6.2.1
six                        1.17.0
smmap                      5.0.2
sniffio                    1.3.1
sounddevice                0.5.1
soundfile                  0.13.1
soupsieve                  2.7
spandrel                   0.4.1
stringzilla                3.12.5
svg.path                   6.3
svglib                     1.5.1
sympy                      1.13.3
tabulate                   0.9.0
tensorrt                   10.10.0.31
tensorrt_cu12              10.10.0.31
tensorrt_cu12_bindings     10.10.0.31
tensorrt_cu12_libs         10.10.0.31
termcolor                  3.1.0
threadpoolctl              3.6.0
tifffile                   2025.3.30
timm                       1.0.15
tinycss2                   1.4.0
tokenizers                 0.21.1
toml                       0.10.2
tools                      1.0.2
torch                      2.7.0+cu126
torchaudio                 2.7.0+cu126
torchsde                   0.2.6
torchvision                0.22.0+cu126
tqdm                       4.67.1
trampoline                 0.1.2
transformers               4.51.3
transparent-background     1.3.3
trimesh                    4.6.8
triton                     3.1.0
triton-windows             3.2.0.post17
typer                      0.15.3
typing_extensions          4.12.2
typing-inspection          0.4.0
tzdata                     2025.2
ultralytics                8.3.131
ultralytics-thop           2.0.14
urllib3                    1.26.20
uv                         0.7.3
vhacdx                     0.0.8.post2
wcwidth                    0.2.13
webcolors                  24.11.1
webencodings               0.5.1
wget                       3.2
wheel                      0.45.1
wrapt                      1.17.2
xxhash                     3.5.0
yacs                       0.1.8
yapf                       0.43.0
yarl                       1.20.0
zipp                       3.21.0

Thanks in advance.

Regards,

offoxx

**Comment by calcuis** [comment]
the nonwritable tensor warning will always be there since tensors in gguf file is nonwritable with torch which is not a problem; btw, run 1.3b or 14b model? 14b model with 4060 might take 15-20min or even longer
